src/wmtk/operations/composite/TetEdgeSwap.cpp

```diff
@@ -0,0 +1,150 @@
+#include "TetEdgeSwap.hpp"
+#include <wmtk/simplex/closed_star.hpp>
+#include <wmtk/simplex/open_star.hpp>
+#include <wmtk/simplex/utils/SimplexComparisons.hpp>
+
+
+#include <wmtk/Mesh.hpp>
+
+namespace wmtk::operations::composite {
+TetEdgeSwap::TetEdgeSwap(Mesh& m, long collapse_index)
+    : Operation(m)
+    , m_split(m)
+    , m_collapse(m)
+    , m_collapse_index(collapse_index)
+{}
+
+std::vector<Simplex> TetEdgeSwap::execute(const Simplex& simplex)
+{
+    const auto split_simplicies = m_split(simplex);
+    if (split_simplicies.empty()) return {};
+    assert(split_simplicies.size() == 1);
+
+    // after split
+    const Tuple& split_ret = split_simplicies.front().tuple();
+    assert(!mesh().is_boundary(split_ret));
+    auto iter_tuple = split_ret;
+    std::vector<Tuple> candidate_edge_tuples;
+    while (true) {
```

Comment:
couldn't calling top dim cofaces work here? i thought we had some order semantic for edges?

Reply:
top_dimension_cofaces does bfs, I guess the order is not what we want